### DISCUSSION TITLE
Sift/AKAZE etc. features

### DISCUSSION THREAD

**Discussion by Cambloid** [stella-cv/stella_vslam#58] [Q&A] [answered]
Are there any plans to add other feature detectors/matchers like sift or akaze etc.?

**Comment by ymd-stella** [comment, marked as answer]
I'm not interested. If matching performance improves overall performance, it's worth a try. Is there any evidence or clues of this?

**Reply by Cambloid** [reply]
I read that sift and akaze is much better when it comes to accuracy but are much slower than orb.

**Reply by surfii3z** [reply]
I recommend [pySLAM](https://github.com/luigifreda/pyslam) for trying out different features.

In my experience, SIFT was way too slow to be used in real-time SLAM.

**Reply by ymd-stella** [reply]
The following was found on https://github.com/introlab/rtabmap/wiki/Change-parameters.

> To not add too much overhead, we use the FAST keypoints detector instead of GFTT used for visual dictionary and odometry. The quality of the keypoints of FAST is lower than with GFTT, but it is very more fast (under 10 ms to extract keypoints).

> With some tests, I've found that GFTT+BRIEF gives the higher reliability across all available binary feature detectors. FAST/ORB detectors are faster than GFTT but they generate less good features to track. FREAK and BRIEF descriptors give similar results.

**Comment by ymd-stella** [comment]
BEBLID (Boosted Efficient Binary Local Image Descriptor) may be a good option.

- https://docs.opencv.org/master/d7/d99/classcv_1_1xfeatures2d_1_1BEBLID.html
- https://github.com/iago-suarez/BEBLID
